docs/source/extension/extension_points.rst

```diff
@@ -721,6 +723,117 @@ When the ``labStatus`` busy state changes, we update the text content of the
 
 .. _widget-tracker:
 
+
+Toolbar Registry
+----------------
+
+As for the context menu and the main menu bar, JupyterLab provides an infrastructure to define
+and customize toolbar widgets of document widgets from the settings. A typical example is
+the notebook toolbar as seen in the snippet below:
```

Comment:
Just a suggestion on making it easier to understand on first read.

```suggestion
JupyterLab provides an infrastructure to define and customize toolbar widgets of document
widgets from the settings, which is similar to that defining the context menu and the main menu
bar. A typical example is the notebook toolbar as in the snippet below:
```

Reply:
Another thought: should "document widgets" be replaced with "`DocumentWidget`s" to make it easier to understand? It might help, but I am not 100% sure if it helps or not.

Reply:
As _document widget_ is not an official term, your suggestion is great. So developers get a hint to look for that word in the code base if they are not aware of that widget type.